src/mrpro/operators/Functional.py

```diff
@@ -0,0 +1,177 @@
+"""Base Class Functional."""
+
+import math
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+
+import torch
+
+from mrpro.operators.Operator import Operator
+
+
+class Functional(Operator[torch.Tensor, tuple[torch.Tensor]]):
+    """Functional Base Class."""
+
+    def _throw_if_negative_or_complex(
+        self, x: torch.Tensor | float, message: str = 'sigma must be real and contain only positive values'
+    ):
+        """Throw an exception if any element of x is negative or complex.
+
+        Raises a ValueError if x contains negative or complex values.
```

Comment:
I would combine this line with the first line as it does not add any additional information except for the ValueError